src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java

```diff
@@ -0,0 +1,110 @@
+package com.breskul.bring;
+
+import com.breskul.bring.exceptions.BeanInitializingException;
+import com.breskul.bring.exceptions.NoSuchBeanException;
+import com.breskul.bring.exceptions.NoUniqueBeanException;
+import com.breskul.bring.exceptions.NuSuchBeanConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Standalone application context, accepting component classes as input — in particular @Configuration-annotated classes,
+ * but also plain @Component types
+ */
+public class AnnotationConfigApplicationContext implements ApplicationContext {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BringApplication.class);
+    private final Map<String, Object> context;
+    private final String packageName;
+
+    public AnnotationConfigApplicationContext(String packageName) {
+        Objects.requireNonNull(packageName);
+        this.context = new ConcurrentHashMap<>();
+        this.packageName = packageName;
+        scan();
+    }
+
+    /**
+     * Scanning a package for classes or methods annotated by @Component and @Bean
+     */
+    private void scan() {
+    }
+
+    /**
+     * Store bean in the context
+     *
+     * @param beanName The name of bean
+     * @param bean     The instance of bean
+     * @param <T>      The generic type of instance
+     */
+    private <T> void storeBean(String beanName, T bean) {
+        Objects.requireNonNull(beanName);
+        Objects.requireNonNull(bean);
+        LOGGER.info("Bring Framework: Creating bean >>> {}: {}", beanName, bean);
+        context.put(beanName, bean);
+    }
+
+    /**
+     * Maps filter by bean type
+     *
+     * @param beanType The Class of type
+     * @param <T>      type of the bean
+     * @return Predicate - filter isAssignableFrom for values types
+     */
+    private <T> Predicate<Map.Entry<String, Object>> filterByBeanType(Class<T> beanType) {
+        return (entry) -> entry.getValue().getClass().isAssignableFrom(beanType);
+    }
+
+    /**
+     * Creating an instance of bean by beanType
+     *
+     * @param beanType - The type of bean instance
+     * @param <T>
+     * @return instance of the created bean
+     */
+    private <T> T createInstance(Class<T> beanType) {
+        var constructor = Arrays.stream(beanType.getConstructors())
+                .findFirst()
+                .orElseThrow(NuSuchBeanConstructor::new);
+        try {
+            T bean = beanType.cast(constructor.newInstance());
+            return bean;
```

Comment:
We can return bean without additional variable. For example:
 `return beanType.cast(constructor.newInstance());`